EAD/Session14/Assignment/MetaParkingSystemV7/src/main/java/com/metacube/metaparkingsystemv2/config/AppSecurityConfig.java

```diff
@@ -0,0 +1,44 @@
+package com.metacube.metaparkingsystemv2.config;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.password.NoOpPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@EnableWebSecurity
+public class AppSecurityConfig extends WebSecurityConfigurerAdapter {
+
+	@Autowired
+	private UserDetailsService userDetailsService;
+	
+	@Override
+	protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+		
+		auth.userDetailsService(userDetailsService);
+	}
+	
+	@Bean
+	public PasswordEncoder getPasswordEncoder() {
+		return NoOpPasswordEncoder.getInstance();
+	}
+
+	@Override
+	protected void configure(HttpSecurity http) throws Exception {
+		
+		http
+			.csrf().disable()
+			.authorizeRequests()
+			.antMatchers("/**").hasAnyRole("USER", "ADMIN")
+			.and()

```

Comment:
How we handle this scenario please explain me and where you handle this

Reply:
Except customSpringSecurityLoginPage.jsp all other pages will lock until login by a authenticate user.